addons/product/models/product.py

```diff
@@ -401,9 +401,9 @@ def _name_get(d):
             name = variant and "%s (%s)" % (product.name, variant) or product.name
             sellers = []
             if partner_ids:
-                sellers = [x for x in product.seller_ids if (x.name.id in partner_ids) and (x.product_id == product)]
```

Comment:
`sellers = product.seller_ids.filtered(lambda x: not x.company_id or x.company_id == self.env.user.company_id)`